packages/core-app-api/src/apis/implementations/auth/microsoft/MicrosoftAuth.ts

```diff
@@ -43,14 +60,108 @@ export default class MicrosoftAuth {
         'email',
         'User.Read',
       ],
+      scopeTransform = scopes => scopes.concat('offline_access'),
     } = options;
 
-    return OAuth2.create({
-      discoveryApi,
-      oauthRequestApi,
-      provider,
-      environment,
-      defaultScopes,
-    });
+    this.environment = environment;
+    this.provider = provider;
+    this.oauthRequestApi = oauthRequestApi;
+    this.discoveryApi = discoveryApi;
+    this.scopeTransform = scopeTransform;
+
+    this.oauth2 = {
+      [MicrosoftAuth.MicrosoftGraphID]: OAuth2.create({
+        discoveryApi: this.discoveryApi,
+        oauthRequestApi: this.oauthRequestApi,
+        provider: this.provider,
+        environment: this.environment,
+        scopeTransform: this.scopeTransform,
+        defaultScopes,
+      }),
+    };
+  }
+
+  private microsoftGraph(): OAuth2 {
+    return this.oauth2[MicrosoftAuth.MicrosoftGraphID];
+  }
+
+  static expectedClaims(scope: string): { aud: string; scp: string } {
+    const scopes = scope.split(' ').map(MicrosoftAuth.parseScope);
+    const firstAudience =
+      scopes.map(({ aud }) => aud).find(aud => aud !== 'openid') ??
+      MicrosoftAuth.MicrosoftGraphID;
+    return {
+      aud: firstAudience,
+      scp: scopes
```

Comment:
I don't see this return value used anywhere, should it be? Or should it be removed

Reply:
It was being used only in tests, but now this function is much simplified and only extracts the resource from a set of scopes